javascript/javascript3/week1/weatherApp.js

```diff
@@ -0,0 +1,112 @@
+    const response = await fetch(`https://api.openweathermap.org/data/2.5/weather?q=${city}&appid=491f420ffa40e14de5218dba143f2809`);
+    const data = await response.json();
+    console.log(data);
+    output.innerHTML = `Temperature in ${city}: ${data.main.temp} </br> icon for the weather type : ${data.weather[0].icon} 
+    </br> The wind speed is :  ${data.wind.speed} </br> the sunrise : ${data.sys.sunrise.getUTCDay} </br>
+    The sunset :  ${data.sys.sunset}   ` ;
+  }
+    catch(error){
+            output.innerHTML = "Error: " + error;
+          };
+  }; 
+
+  button.addEventListener("click", function() {
+    
+    getWeatherData(cityName.value);
+
+  });
+
+
+  //Use my current position optional
+  const butn= document.getElementById('getLocation')
+  const result =document.getElementById('result'); 
+
+const  getlocation=  function() {
+      // Check if Geolocation API is supported by the browser
+      if ("geolocation" in navigator) {
+        navigator.geolocation.getCurrentPosition(function (position) {
+          const latitude = position.coords.latitude;
+          const longitude = position.coords.longitude;
+
+          // Fetch weather data from the position
+          fetch(`https://api.openweathermap.org/data/2.5/weather?lat=${latitude}&lon=${longitude}&appid=491f420ffa40e14de5218dba143f2809`)
+            .then(response => response.json())
+            .then(data => console.log(data))
+            .catch(error => console.error(error));
+
+            result.innerHTML = " This is Latitude: " + latitude + 
+            "<br> This is Longitude: " + longitude;
+        });
+      } else {
+        console.error("Geolocation API is not supported by this browser.");
+      }
+        initMap();
+    };
+
+
+
+let map;
+
+function initMap() {
+  map = new google.maps.Map(document.getElementById("map"), {
+    center: { lat: -34.397, lng: 150.644 },
+    zoom: 8,
+  });
+}
+
+window.initMap = initMap;
+
+const API_KEY = "88jWMys0Mhd9cQ7s0zGC6MsnyNNHK5KB";
+const form = document.querySelector("#gif-search-form");
+const input = document.querySelector("#gif-search-input");
+const resultsDiv = document.querySelector("#gif-results");
+
+form.addEventListener("submit", function(event) {
+  event.preventDefault();
+  const searchTerm = input.value;
+  searchGifs(searchTerm);
+});
+
+function searchGifs(searchTerm) {
+  const endpoint = `https://api.giphy.com/v1/gifs/search?api_key=${API_KEY}&q=${searchTerm}&limit=15`;
+
+  fetch(endpoint)
+    .then(function(response) {
+      return response.json();
+    })
+    .then(function(data) {
+      const gifs = data.data;
+      displayGifs(gifs);
+    })
+    .catch(function(error) {
+      console.error("Error fetching GIFs:", error);
+    });
+}
+
+function displayGifs(gifs) {
+  resultsDiv.innerHTML = "";
+  gifs.forEach(function(gif) {
+    const gifImg = document.createElement("img");
+    gifImg.src = gif.images.fixed_height.url;
+    resultsDiv.appendChild(gifImg);
+  });
+}
```

Comment:
nice job . 
it also nice to give the user a message if search term is empty.

Reply:
thanks i will do that 👍